run-checks

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash -e
+
+# shellcheck disable=SC2145
+
+#----------------------------------------------------#
+# PRINTING ROUTINES                                  #
+#----------------------------------------------------#
+
+P_RESET="\e[0m"
+P_RED="\e[31m"
+P_GREEN="\e[32m"
+
+print_newline() {
+	echo -e "$P_RESET"
+}
+
+print_normal() {
+	echo -e "$P_RESET$@"
+}
+
+print_red() {
+	echo -e "$P_RED$@$P_RESET"
+}
+
+print_green() {
+	echo -e "$P_GREEN$@$P_RESET"
+}
+
+print_special() {
+	echo -ne "$P_RESET$@"
+}
+
+print_err() {
+	print_newline
+	pr "Error: $1"
+	print_newline
+	exit 1
+}
+
+#----------------------------------------------------#
+# TEST OUTPUT CONTROL                                #
+#----------------------------------------------------#
+
+print_test() {
+	print_special "$@"
+}
+
+run() {
+        if $VERBOSE; then
+                v=$(exec 2>&1 && set -x && set -- "$@")
+		echo
+                echo "#${v#*--}"
+		set +e
+                "$@"
+		RET=$?
+		set -e
+        else
+		set +e
+                "$@" >/dev/null 2>&1
+		RET=$?
+		set -e
+        fi
+}
+
+return_check() {
+	if [ $RET -ne 0 ]; then
+                print_newline
+                print_red "Error: $1 (run 'run-checks --verbose' for details)"
+                print_newline 
+                exit 1
+	else
+		print_green "Pass"
+	fi
+}
+
+#----------------------------------------------------#
+# CMDLINE OPTIONS                                    #
+#----------------------------------------------------#
+
```

Comment:
Just out of curiosity, why not `getopt` here?

Reply:
Removed the need for options. Looks way cleaner now.